streams/src/main/java/org/apache/kafka/streams/state/internals/MemoryLRUCacheBytesEntry.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+import org.apache.kafka.streams.processor.RecordContext;
+
+/**
+ * A cache entry
+ */
+public class MemoryLRUCacheBytesEntry implements RecordContext {
```

Comment:
Since this class is referenced in public `MemoryLRUCacheBytes.put`, we need to make it a public interface as well.

Or, we can still make it internal while changing `MemoryLRUCacheBytes` interfaces to

```
void put(String namesapce, byte[] key, byte[] value, RecordContext context);

byte[] get(String namespace, byte[] key, RecordContext context /*this reference will be updated*/);
```

And then this class only need to keep track of `sizeInBytes` and `isDirty` and hence can be an internal class.